icu4c/source/common/putil.cpp

```diff
@@ -1171,16 +1171,16 @@ uprv_tzname(int n)
         because the tzfile contents is underspecified.
         This isn't guaranteed to work because it may not be a symlink.
         */
-        int32_t ret = (int32_t)readlink(TZDEFAULT, gTimeZoneBuffer, sizeof(gTimeZoneBuffer)-1);
-        if (0 < ret) {
+        char *ret = realpath(TZDEFAULT, gTimeZoneBuffer);
```

Comment:
Should the preceding comment be adjusted?

Reply:
No, what that comment says would still be true after these changes.